main.go

```diff
@@ -4,21 +4,30 @@ import (
 	"flag"
 	"net/http"
 	"os"
+	"regexp"
+	"strings"
 	"time"
 
 	"github.com/utilitywarehouse/kube-service-mirror/kube"
 	"github.com/utilitywarehouse/kube-service-mirror/log"
+	"k8s.io/client-go/kubernetes"
 )
 
 var (
 	flagKubeConfigPath       = flag.String("kube-config", "", "Path of a kube config file, if not provided the app will try to get in cluster config")
-	flagTargetKubeConfigPath = flag.String("target-kube-config", "", "(required) Path of the target cluster kube config file to mirrot services from")
+	flagTargetKubeConfigPath = flag.String("target-kube-config", "", "Path of the target cluster kube config file to mirrot services from")
 	flagLogLevel             = flag.String("log-level", "info", "Log level, defaults to info")
 	flagResyncPeriod         = flag.Duration("resync-period", 60*time.Minute, "Namespace watcher cache resync period")
 	flagMirrorNamespace      = flag.String("mirror-ns", "", "The namespace to create dummy mirror services in")
 	flagSvcPrefix            = flag.String("svc-prefix", "", "(required) A prefix to apply on all mirrored services names. Will also be used for initial service sync")
 	flagLabelSelector        = flag.String("label-selector", "", "(required) Label of services and endpoints to watch and mirror")
 	flagSvcSync              = flag.Bool("svc-sync", true, "sync services on startup")
+
+	saToken = os.Getenv("SERVICE_ACCOUNT_TOKEN")
+	apiURL  = os.Getenv("KUBE_API_SERVER")
+	caURL   = os.Getenv("REMOTE_CLUSTER_CA_CERT_URL")
```

Comment:
```suggestion
	saToken = os.Getenv("REMOTE_SERVICE_ACCOUNT_TOKEN")
	apiURL  = os.Getenv("REMOTE_KUBE_API_SERVER")
	caURL   = os.Getenv("REMOTE_CLUSTER_CA_CERT_URL")
```
Prefix all of these values with `REMOTE` to help communicate their usage?

Also, I don't like the inconsistent mixture of env vars and flags. I'd prefer flags for everything, env vars or both for all options. The token could be a path to a file so it isn't exposed in the flags?